lib/taskana-core/src/test/java/acceptance/classification/CreateClassificationAccTest.java

```diff
@@ -1,297 +1,374 @@
+    Classification classification = classificationService.newClassification("Key7", "", "TASK");
+    classification = classificationService.createClassification(classification);
 
-    classification = CLASSIFICATION_SERVICE.newClassification("newKey71", "", "TASK");
-    classification.setServiceLevel("");
-    classification = CLASSIFICATION_SERVICE.createClassification(classification);
     assertThat(classification.getServiceLevel()).isEqualTo("P0D");
+    assertThat(classification.getId()).isNotEqualTo(null);
+    assertThat(classification.getId()).isNotEmpty();
+    assertThat(classification.getCreated()).isNotNull();
+    assertThat(classification.getModified()).isNotNull();
+    assertThat(classification.getParentId()).isEqualTo("");
+    assertThat(classification.getParentKey()).isEqualTo("");
+    assertThat(classification.getIsValidInDomain()).isEqualTo(false);
   }
 
-  @WithAccessId(user = "admin")
-  @Test
-  void should_SetDefaultServiceLevel_When_TryingToUpdateClassificationWithMissingServiceLevel()
-      throws Exception {
-    Classification classification =
-        CLASSIFICATION_SERVICE.getClassification("CLI:000000000000000000000000000000000001");
-    classification.setServiceLevel(null);
-    classification = CLASSIFICATION_SERVICE.updateClassification(classification);
-    assertThat(classification.getServiceLevel()).isEqualTo("P0D");
+  @WithAccessId(user = "businessadmin")
+  @AfterAll
+  void should_NotCreateClassification_When_ExceptionIsThrownDuringCreation() {
+    List<ClassificationSummary> classifications =
+        classificationService.createClassificationQuery().keyIn("KeyErrCreation").list();
 
-    classification.setServiceLevel("");
-    classification = CLASSIFICATION_SERVICE.updateClassification(classification);
-    assertThat(classification.getServiceLevel()).isEqualTo("P0D");
+    assertThat(classifications).isEmpty();
+  }
+
+  @Nested
+  @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+  class CreatingChildClassificationTest {
+    String parentId;
+    String parentKey;
+
+    @WithAccessId(user = "businessadmin")
+    @BeforeAll
+    void setupParentClassification() throws Exception {
+      ClassificationSummary parentClassification =
+          DefaultTestEntities.defaultTestClassification()
+              .key("Key0")
+              .domain("DOMAIN_A")
+              .type("TASK")
+              .buildAndStoreAsSummary(classificationService);
+      parentId = parentClassification.getParentId();
+      parentKey = parentClassification.getParentKey();
+    }
+
+    @WithAccessId(user = "businessadmin")
+    @Test
+    void should_CreateClassification_WhenCreatingChildClassification() throws Exception {
+      Classification childClassification =
+          classificationService.newClassification("Key0_1", "DOMAIN_A", "TASK");
+      childClassification.setParentId(parentId);
+      childClassification.setParentKey(parentKey);
```

Comment:
Our API is not supposed to be used with both. Please create a test case for eather.
They could be named `should_CreateChildClassification_WhenAssigningParentKey' and `should_CreateChildClassification_WhenAssigningParentId'

Reply:
Challenge: Do that in a TestFactory ;)

Reply:
Challenge accepted :) How about this: 
```
    @WithAccessId(user = "businessadmin")
    @TestFactory
    Stream<DynamicTest> should_CreateMasterClassification_WhenCreatingChildClassification()
        throws Exception {
      List<Consumer<Classification>> setterList = new ArrayList<>();
      setterList.add(p -> p.setParentKey(parentKey));
      setterList.add(p -> p.setParentId(parentId))
      Iterator<Consumer<Classification>> iterator = setterList.iterator();

      AtomicInteger i = new AtomicInteger();

      ThrowingConsumer<Consumer<Classification>> test =
          consumer -> {
            Classification childClassification =
                classificationService.newClassification("Key0_2_" + i.get(), "DOMAIN_A", "TASK");
            consumer.accept(childClassification);
            classificationService.createClassification(childClassification);

            ClassificationImpl expectedMasterClassification =
                (ClassificationImpl) childClassification.copy("Key0_2_" + i.get());
            expectedMasterClassification.setDomain("");
            consumer.accept(expectedMasterClassification);
            List<ClassificationSummary> classifications =
                classificationService
                    .createClassificationQuery()
                    .keyIn("Key0_2_" + i.getAndIncrement())
                    .list();
            assertThat(classifications)
                .usingRecursiveFieldByFieldElementComparatorIgnoringFields("id")
                .contains(expectedMasterClassification.asSummary());
          };

      return DynamicTest.stream(iterator, c -> String.format("for '%s'", i.get()), test);
    }
```